OSDOCS 10841: Added helm charts updates in oc-mirror v2 docs. (#86685)
openshift/openshift-docs: modules/oc-mirror-imageset-config-parameters-v2.adoc

|Array of strings 
Example: `docker.io/library/alpine`

|`mirror.helm`

> |The helm configuration of the image set. Note that the oc-mirror plugin supports only helm charts that do not require user input when rendered.

Reviewer: ```suggestion
|The helm configuration of the image set. Note that the oc-mirror plugin does not support helm charts which requires changes to the `values.yaml` to render.
```

|Object

|`mirror.helm.local`
|The local helm charts to mirror.
|Array of objects. For example:

[source,yaml]
----
local:
  - name: podinfo
    path: /test/podinfo-5.0.0.tar.gz
----

|`mirror.helm.local.name`
|The name of the local helm chart to mirror.
|String. For example: `podinfo`.

|`mirror.helm.local.path`
|The path of the local helm chart to mirror.
|String. For example: `/test/podinfo-5.0.0.tar.gz`.

|`mirror.helm.repositories`
|The remote helm repositories to mirror from.
|Array of objects. For example:

[source,yaml]

> ----
> repositories:
>   - name: podinfo
>     url: https://example.github.io/podinfo
>     charts:
>       - name: podinfo
>         version: 5.0.0
> ----

Reviewer: ```suggestion
----
repositories:
  - name: podinfo
    url: https://example.github.io/podinfo
    charts:
      - name: podinfo
        version: 5.0.0
         imagePaths:
         - "{.spec.template.spec.custom[*].image}"
----
```

|`mirror.helm.repositories.name`
|The name of the helm repository to mirror from.
|String. For example: `podinfo`.

|`mirror.helm.repositories.url`
|The URL of the helm repository to mirror from.
|String. For example: [x-]`https://example.github.io/podinfo`.

|`mirror.helm.repositories.charts`
|The remote helm charts to mirror.
|Array of objects.

|`mirror.helm.repositories.charts.name`
|The name of the helm chart to mirror.
|String. For example: `podinfo`.

|`mirror.helm.repositories.charts.version`
|The version of the named helm chart to mirror.
|String. For example: `5.0.0`.

> 

Reviewer: ```suggestion
|`mirror.helm.repositories.charts.imagePaths`
|The custom path of a container image inside of the helm chart. Note: oc-mirror will detect and mirror container images of the helm chart searching on well-known paths, but custom paths can be provided on this field.
|Array of string. For example: "- {.spec.template.spec.custom[*].image}".
```

|Array of strings 
Example: `docker.io/library/alpine`

|`mirror.helm`

> |The helm configuration of the image set. Note that `oc-mirror` does not support helm charts with manually modified `values.yaml`.

Reviewer 2: @subhtk - yes this is much better - thanks !!!

Author: Great.

> |The helm configuration of the image set. Note that `oc-mirror` does not support helm charts with manually modified `values.yaml`.

Reviewer 3: ```suggestion
|The helm configuration of the image set. Note that `oc-mirror` does not support helm charts with manually modified `values.yaml` files.
```
OR

The helm configuration of the image set. Note that `oc-mirror` does not support helm charts with a manually modified `values.yaml` file.

> |The helm configuration of the image set. Note that `oc-mirror` does not support helm charts with manually modified `values.yaml`.

Reviewer 4: A slight addition to @sr1kar99 's suggestion
```suggestion
|The helm configuration of the image set. The `oc-mirror` plugin does not support helm charts with manually modified `values.yaml` files.
```

> |The helm configuration of the image set. Note that the oc-mirror plugin does not support helm charts which requires changes to the `values.yaml` to render.

Reviewer 2: @aguidirh @subhtk - Honestly this is not very clear to me.  Could we elaborate more or add a more detailed note and point to the note ?

Reviewer: Hi @lmzuccarelli,

This piece of documentation is to explain what's is the mirror.helm field in the ImageSetConfiguration:

```
|`mirror.helm`
|The helm configuration of the image set. Note that the oc-mirror plugin does not support helm charts which requires changes to the `values.yaml` to render.
```

It follows the same standard of the mirror.operators field in the ImageSetConfiguration:
```
|`mirror.operators`
|The Operators configuration of the image set.
|Array of objects 
```

The only difference is the note where it tries to explain that helm charts that depending on rendering the `values.yaml` in order to discover the images are not supported.

If you have a suggestion on how to make it clearer feel free to suggest.

Reviewer 2: @aguidirh - so this to me makes more sense as you mentioned _"The only difference is the note where it tries to explain that helm charts that depending on rendering the values.yaml in order to discover the images are not supported" 

@subhtk - can you wordsmith that sentence a bit ?

> |The helm configuration of the image set. Note that the oc-mirror plugin does not support helm charts which requires changes to the `values.yaml` to render.

Reviewer: ```suggestion
|The helm configuration of the image set. Note oc-mirror does not support manually changed `values.yaml`.
```

|Object

|`mirror.helm.local`
|The local helm charts to mirror.
|Array of objects. For example:

[source,yaml]
----
local:
  - name: podinfo
    path: /test/podinfo-5.0.0.tar.gz
----

|`mirror.helm.local.name`
|The name of the local helm chart to mirror.
|String. For example: `podinfo`.

|`mirror.helm.local.path`
|The path of the local helm chart to mirror.
|String. For example: `/test/podinfo-5.0.0.tar.gz`.

|`mirror.helm.repositories`
|The remote helm repositories to mirror from.
|Array of objects. For example:

[source,yaml]
----
repositories:
  - name: podinfo
    url: https://example.github.io/podinfo
    charts:
      - name: podinfo
        version: 5.0.0
         imagePaths:

>          - "{.spec.template.spec.custom[*].image}"

Reviewer 5: @aguidirh is the `imagePaths` here are required ? I never tried mirroring helm charts with this option. My imageSetConfig always has been as below
```
apiVersion: mirror.openshift.io/v2alpha1
kind: ImageSetConfiguration
mirror:
  helm:
    repositories:
      - name: podinfo
        url: https://stefanprodan.github.io/podinfo
        charts:
          - name: podinfo
            version: 5.0.0
      - name: sbo
        url: https://redhat-developer.github.io/service-binding-operator-helm-chart/
```

Reviewer: Hi @kasturinarra,

This field is available but I do not know a helm chart with a customized image path that allow us to test it. 

I decided to keep it because in v1 we have this field available as well, so I am not sure if there are customers using it.

Reviewer 5: @aguidirh if that is the case lets keep it to be compatible with v1 may be ? WDYS ?

Reviewer: @kasturinarra I agree. In this ways it is easier for customers using v1 to migrate to v2.
